R/functions_io.R

```diff
@@ -1058,8 +1057,12 @@ ReadCounts = function(path, technology, assays, barcode_metadata=NULL, feature_m
   
   # Make feature names Seurat-compatible (replace '_' with '-') and unique
   for(i in seq_along(counts_lst)) {
```

Comment:
I agree and I can only recommend to have a look at the purrr::map functions which are improved versions of the apply family and work (in my opinion) smoother.

However, sometimes for loops are the better alternative. But I have added an issue #43 so that we can over time track these cases and decide what to do. For this pull request, I would not do it since it takes some time.